### PR TITLE
Feat: system enrollment error - 500 error during /token

### PR DESCRIPTION
**Pull request by angela-tran**
Part of #2032 

 - [x] 500-level error that occurs during `/token` returns a JSON response that is used to send user to `/enrollment/error`
 - [x] `/enrollment/error` shows system enrollment error template
 - [x] Sentry notification is sent
 - [x] Analytic event is sent (`failed access token request` with `status_code` in `event_properties`)

### Reviewing / testing
Similar to review steps for #2088, you can add some code locally to simulate a 500 error.
To `/token`, add:
```diff
        try:
+          class response:
+             def __init__(self):
+                self.status_code = 500
+
+          raise HTTPError(response=response())
           response = client.request_card_tokenization_access()
        except Exception as e:
```

### CODE REVIEW

File: benefits/enrollment/views.py [hunk 1]
```diff
@@ -45,8 +46,20 @@ def token(request):
...
         )
         client.oauth.ensure_active_token(client.token)
-        response = client.request_card_tokenization_access()
-        session.update(request, enrollment_token=response.get("access_token"), enrollment_token_exp=response.get("expires_at"))
+
+        try:
+            response = client.request_card_tokenization_access()
+        except Exception as e:
+            if isinstance(e, HTTPError) and e.response.status_code >= 500:
+                sentry_sdk.capture_exception(e)
```
Comment (angela-tran) [inline benefits/enrollment/views.py:54]: Send a new specific analytic event for us failing to get an access token for card tokenization

File: benefits/enrollment/urls.py [hunk 2]
```diff
@@ -15,4 +15,5 @@
     path("reenrollment-error", views.reenrollment_error, name="reenrollment-error"),
     path("retry", views.retry, name="retry"),
     path("success", views.success, name="success"),
+    path("system_error", views.system_error, name="system_error"),
```
Comment (angela-tran) [inline benefits/enrollment/urls.py:18]:
```suggestion
    path("error", views.system_error, name="system_error"),
```

File: benefits/enrollment/views.py [hunk 3]
```diff
@@ -45,8 +46,20 @@ def token(request):
...
+                data = {"redirect": reverse(ROUTE_SYSTEM_ERROR)}
+                return JsonResponse(data)
+            else:
+                raise e
```
Comment (angela-tran) [inline benefits/enrollment/views.py:58]: Send the same analytic event class as above. It should include the error code.

**Comment by thekaveman** [comment]
Rebased on `dev` to incorporate changes from #2088

File: benefits/enrollment/views.py (hunk 3 above)
Comment (angela-tran) [reply benefits/enrollment/views.py:58]: Will be done for 400-level error case of `/token` (see #2032 acceptance criteria)

File: benefits/enrollment/views.py (hunk 1 above)
Comment (angela-tran) [reply benefits/enrollment/views.py:54]: See [line 53](https://github.com/cal-itp/benefits/pull/2089/files#diff-b67985b5acabad774a30a1fccbb912e611d4056d38316cd6614ce601f6930335R53)

**Review by thekaveman — APPROVED** [review]
This looks great. I made the suggested modification in the view and was sent to the expected error page when trying to acquire the access token.

File: benefits/enrollment/templates/enrollment/index.html [hunk 4]
```diff
@@ -41,6 +41,12 @@ <h1 class="pb-lg-8 pb-4">
       $.ajax({ dataType: "script", attrs: { nonce: "{{ request.csp_nonce }}"}, url: "{{ card_tokenize_url }}" })
           .done(function() {
           $.get("{{ access_token_url }}", function(data) {
+              if (data.redirect) {
+                // https://stackoverflow.com/a/42469170
+                // use 'assign' because 'replace' was giving strange Back button behavior
+                window.location.assign(data.redirect);
```
Comment (thekaveman) [inline benefits/enrollment/templates/enrollment/index.html:47]: I'm not worried about this as blocking this PR, but...

I wonder if `window.location.replace` is the right method here instead? Do we want to support the back button when the user is supposed to wait and not try again immediately?

Comment (angela-tran) [reply benefits/enrollment/templates/enrollment/index.html:47]: I created #2119 to follow up on this

**Pull request merged**